src/genotype.cpp

```diff
@@ -59,10 +60,15 @@ Genotype::Genotype(const int inputs, const int outputs){
                 }
         }
 
-        /**
-         * FIXME: After adding the graph representation of the network,
-         * FIXME: also construct the graph HERE!
-         */
+        // construct graph representation of the initial network
```

Comment:
Note the similarities between this graph construction and the next graph construction: they can both be constructed from connection_genes. So why not create a helper method
`void construct_graph(const std::list<Connection>& connections);` so that you don't have to type the same thing twice.